File: lidar_node.py
#!/usr/bin/env python3
#coding=utf-8
import rospy
from sensor_msgs.msg import LaserScan
from geometry_msgs.msg import Twist
cnt=0
def LidarCallback(msg):
    global cnt
    global vel_pub
    dist=msg.ranges[180]
    rospy.loginfo("前方测距range[180]=%f米",dist)
    if(cnt>0):
        cnt=cnt-1
        return    
    vel_cmd=Twist()

    if dist<1.5:
        vel_cmd.angular.z=0.3
        cnt=50
    else:    
        vel_cmd.linear.x=0.1
    vel_pub.publish(vel_cmd)
if __name__=="__main__":
    rospy.init_node("lidar_node")
    lidar_sub=rospy.Subscriber("/scan",LaserScan,LidarCallback,queue_size=10)
    vel_pub=rospy.Publisher("/cmd_vel",Twist,queue_size=10)
    rospy.spin()

File: lidar_node.cpp
#include<ros/ros.h>
#include<sensor_msgs/LaserScan.h>
#include<geometry_msgs/Twist.h>
ros::Publisher vel_pub;
int cnt;
void LidarCallBack(const sensor_msgs::LaserScan msg)
{
    float fMidDist=msg.ranges[180];
    ROS_INFO("前方测距range[180]=%f米",fMidDist);
    geometry_msgs::Twist vel_cmd;
    if(cnt>0)
    {
        cnt--;
        return;
    }
    if(fMidDist<1.5)
        vel_cmd.angular.z=0.3,cnt=50;
    else
        vel_cmd.linear.x=0.1;
    vel_pub.publish(vel_cmd);
}
int main(int argc,char *argv[])
{
    setlocale(LC_ALL,"");
    ros::init(argc,argv,"lidar_node");
    ros::NodeHandle n;
    ros::Subscriber lidar_sub=n.subscribe("/scan",10,&LidarCallBack);
    
    vel_pub=n.advertise<geometry_msgs::Twist>("/cmd_vel",10);
    ros::spin();
    return 0;
}